dev-docs/RFCs/vNext/text-layer-font-setting-api.md

```diff
@@ -0,0 +1,63 @@
+# RFC: Text Layer API Change
+
+* **Authors**: Xintong Xia 
+* **Date**: January 2019
+* **Status**: Draft 
+
+## Motivation 
+
+When `TextLayer` generates shared `fontAtlas` for required `characterSet`, it applied default settings, like 
+`fontSize`, `buffer`(Whitespace surround each character), etc.. However one setting can not be suitable for 
+all the different fonts. For example, [Cinzel](https://fonts.google.com/specimen/Cinzel) has a special shape 
+of `Q` character, which requires a big buffer [See detail](https://github.com/uber/deck.gl/pull/2609). Big buffer around each character can be very expensive 
+if the target `characterSet` is considerably large, which is not necessary for other fonts.
+
+`TextLayer` needs expose settings related to `fontAtlas` generation for users to manipulate when needed.
+
+## Proposal: Expose font settings as layer props
+
+`TextLayer` supports [`sdf`](https://github.com/mapbox/tiny-sdf), check [Text Layer](/docs/layers/text-layer.md) for details.
+
+`fontSettings` together with the other two layer props `characterSet` and `fontFamily` will be used in generating `fontAtlas`.
+
+```js
+
+import {TextLayer} from '@deck.gl/layers'
+
+const textLayer = new TextLayer({
+  ...,
+  characterSet: 'abcdefg',
+  fontFamily: 'Monaco, monospace',
+
+  fontSettings: {
+    // shared options between non-sdf and sdf
+    // this fontSize if only applied for generating fontAtlas
+    // it does not impact the size of the text labels 
+    fontSize: 64, 
```

Comment:
Need a better name to avoid confusion with the actual display size (i.e. `getSize` and `sizeScale` props)